docs/proposals/implement-api-that-imports-prometheus-blocks-into-cortex.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Implement API that imports Prometheus blocks into Cortex"
+linkTitle:
+weight:
+slug:
+---
+
+- Author: [Siddharth Asthana](https://github.com/edith007)
+- Date: March 2023
+- Status: draft
+
+## Problem statement
+
+There might be a lot of cases where the users might already be using Prometheus and later decides to switch to Cortex. Cortex will take care of the metrics which will be generated after the switch, but at present there is no way to backfill the old Prometheus TSDB blocks to Cortex. This proposal is to add the ability in cortex to fetch the Prometheus TSDB blocks and backfill the data.
+
+## Detailed Solution
+
+We can create a new service called `Importer` which will be responsible for bringing in the old Prometheus TSDB blocks to Cortex. 
+
+### How Importer will work
+
+The importer service will have two API endpoints: `import` and `import/status/{id}`
+
+`import`:
+
+- It will take a link to where the old Prometheus TSDB blocks are stored
```

Comment:
Then that means importer or Cortex needs to access the link? I feel this is not usually the case. 
Most of the time client side can talk to Cortex but not vice versa. Take a local TSDB block as an example.

Reply:
I think i can see pros and cons on both solutions.

But i think im now leaning towards options 2 (having a cortex CLI that uploads blocks by blocks and show a progress bar seems nice - we will not need to have the status API as this would be the feedback from the CLI?)

Reply:
Talking a link that cortex should pull sounds nice, but how is that link authorized? how long is the client giving cortex access?. How can the client cut that access if they need too?. what kind of secrets should cortex receive to pull that data? How should client debug problems when client can access data but cortex can't?  How many type of auths will need to be supported? The timeouts mentioned can still happen between cortex and user storage, only this time user can't make a decision directly.

These problems are faster to solve in a cortex-cli, as clients can tweak their cli as needed.

Reply:
I also agree with @alanprot  👍🏻 I am also leaning towards the option 2 ✋🏻 So think we can make a CLI tool… But there are other things to worry about, which @friedrichg said ☝

Reply:
So everybody agree with option 2? @yeya24 @friedrichg @alvinlin123 

If so lets update the proposal?

Reply:
agree with option 2.

Reply:
I like a separate CLI, too. Is this part of this repo or part of `cortex-tools`?
If we go with CLI, does it kind of conflict with the status API we use to get the job status asyncly?

Reply:
Yeah.. i think if we go block by block the status API is not needed, right?